mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractMailboxManagerAttachmentTest.java

```diff
@@ -0,0 +1,197 @@
+        assertThat(messages.hasNext()).isTrue();
+        assertThat(messages.next().getAttachmentsIds()).hasSize(1);
+    }
+
+    @Test
+    public void appendMessageShouldNotStoreAnyAttachmentWhenUnparsableMail() throws Exception {
+        MailboxManager parseFailingMailboxManager = getParseFailingMailboxManager();
+        MessageManager parseFailingInboxMessageManager = parseFailingMailboxManager.getMailbox(inboxPath, mailboxSession);
+        InputStream mailInputStream = new ByteArrayInputStream("content".getBytes());
+        parseFailingInboxMessageManager.appendMessage(mailInputStream, SUN_SEP_9TH_2001, mailboxSession, true, new Flags(Flag.RECENT));
+
+        Iterator<MailboxMessage> messages = messageMapper.findInMailbox(inbox, MessageRange.all(), FetchType.Full, 1);
+        assertThat(messages.hasNext()).isTrue();
+        List<AttachmentId> attachmentsIds = messages.next().getAttachmentsIds();
+        assertThat(attachmentsIds).hasSize(0);
+    }
+
+    @Test
+    public void appendMessageShouldStoreOnceWhenDuplicateAttachment() throws Exception {
+        InputStream mailInputStream = ClassLoader.getSystemResourceAsStream("eml/oneAttachmentAndSomeInlined.eml");
+        InputStream mailInputStream2 = ClassLoader.getSystemResourceAsStream("eml/oneAttachmentAndSomeInlined.eml");
+        String user2 = "user2@domain.tld";
+        MailboxSession user2MailboxSession = new MockMailboxSession(user2);
+        MessageMapper user2MessageMapper = getMailboxSessionMapperFactory().getMessageMapper(user2MailboxSession);
+        MailboxMapper user2MailboxMapper = getMailboxSessionMapperFactory().getMailboxMapper(user2MailboxSession);
+        MailboxPath user2InboxPath = new MailboxPath(MailboxConstants.USER_NAMESPACE, user2, "INBOX");
+        mailboxManager.createMailbox(user2InboxPath, user2MailboxSession);
+        Mailbox user2Inbox = user2MailboxMapper.findMailboxByPath(user2InboxPath);
+        MessageManager user2InboxMessageManager = mailboxManager.getMailbox(user2InboxPath, user2MailboxSession);
+
+        inboxMessageManager.appendMessage(mailInputStream, SUN_SEP_9TH_2001, mailboxSession, true, new Flags(Flags.Flag.RECENT));
+        user2InboxMessageManager.appendMessage(mailInputStream2, SUN_SEP_9TH_2001, user2MailboxSession, true, new Flags(Flags.Flag.RECENT));
+
+        Iterator<MailboxMessage> messages = messageMapper.findInMailbox(inbox, MessageRange.all(), FetchType.Full, 1);
+        Iterator<MailboxMessage> user2Messages = user2MessageMapper.findInMailbox(user2Inbox, MessageRange.all(), FetchType.Full, 1);
+        assertThat(messages.hasNext()).isTrue();
+        List<AttachmentId> attachmentsIds = messages.next().getAttachmentsIds();
+        assertThat(attachmentsIds).hasSize(1);
+        assertThat(user2Messages.hasNext()).isTrue();
+        List<AttachmentId> user2AttachmentsIds = user2Messages.next().getAttachmentsIds();
+        assertThat(attachmentsIds.equals(user2AttachmentsIds)).isTrue();
+    }
```

Comment:
Could you add some tests with contentType != Full ?

Just to make sure attachements are not retrieved....